core/client/src/main/java/alluxio/client/file/FileOutStream.java

```diff
@@ -198,21 +205,12 @@ public void close() throws IOException {
         if (mCanceled) {
           // TODO(yupeng): Handle this special case in under storage integrations.
           mUnderStorageOutputStream.close();
-          if (!ufs.exists(tmpPath)) {
-            // Location of the temporary file has changed, recompute it.
-            updateUfsPath();
-            tmpPath = PathUtils.temporaryFileName(mNonce, mUfsPath);
-          }
           ufs.delete(tmpPath, false);
         } else {
           mUnderStorageOutputStream.flush();
           mUnderStorageOutputStream.close();
-          if (!ufs.exists(tmpPath)) {
-            // Location of the temporary file has changed, recompute it.
-            updateUfsPath();
-            tmpPath = PathUtils.temporaryFileName(mNonce, mUfsPath);
-          }
-          if (!ufs.rename(tmpPath, mUfsPath)) {
+          if (!ufs.rename(tmpPath, mUfsPath)) { // Failed to commit file
+            ufs.delete(tmpPath, false);
```

Comment:
do we have this logic to delete ufs tmp file before? Are we changing the behavior on failure cases after this PR?